### PR TITLE
Recipe app: Group Projects

### PR DESCRIPTION
**Pull request by amrendrakind**
### Recipe app: Project requirements
- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    - Should be built with Devise.
- Food list:
   - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
   - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
     - Should display a list of all public recipes ordered by newest as in the wireframe.
     - Should lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
   - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
   - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.

**Unit and Integration Tests**
Screenshot from 2022-08-06 13-57-52

### CODE REVIEW

File: app/views/foods/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                </tr>
+            </thead>
+            <tbody>
+            <% @foods.each do |food| %>
+                <tr>
+                    <td><%=food.name %></td>
+                    <td><%=food.measurement_unit%></td>
+                    <td><%=food.price%></td>
+                    <td><%=food.quantity%></td>
+                    <td><%=button_to 'Delete', user_food_path(@user, food), method: :delete, class:'remove-recipe-button'%></td>
```
Comment (Deepakdanger) [inline app/views/foods/index.html.erb]: - [ ] I can also delete a food item from the lists created by different users, but owners should only do it.

File: app/views/recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      <div class='food-add-button'>
+        <%= button_to link_to 'Add Recipe', new_user_recipe_path %>
+      </div>
+
+      <ul class="recipes-list">
+        <% @recipes.each do |recipe| %>
+          <li>
+              <div class="label">
+                  <b><%= link_to recipe.name , user_recipe_path(@user, recipe) %></b>
+                  <%= button_to 'Remove', user_recipe_path(@user, recipe), method: :delete, class: 'remove-recipe-button'%>
```
Comment (Deepakdanger) [inline app/views/recipes/index.html.erb]: - [ ] Same with recipes. I can also delete recipes from the lists created by different users, but owners should only do it.

File: app/views/recipes_foods/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <tbody>
+              <% @foods.each do |food| %>
+                <tr>
+                  <%= form_with model: @recipe_food, url: user_recipe_recipes_foods_path do |f| %> 
+                  <td>  <%= food.name %></td>
+                  <td> <%= food.measurement_unit %></td>
+                  <td> <%= food.price %></td>
+                  <td>  <%= f.number_field :quantity %></td>
+                    <%= f.hidden_field :food_id, value: food.id %>
+                    <td>  <%= f.submit 'Add Items' %></td>
```
Comment (Deepakdanger) [inline app/views/recipes_foods/new.html.erb]: - [ ] I can't able to add ingredients. when I click on `add items`, nothing happened, Please fix this.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @amrendrakind, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required changes
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/recipes_foods/new.html.erb (hunk 3 above)
Comment (amrendrakind) [reply app/views/recipes_foods/new.html.erb]: @Deepakdanger Thank you for your feedback.

File: app/views/foods/index.html.erb (hunk 1 above)
Comment (amrendrakind) [reply app/views/foods/index.html.erb]: @Deepakdanger, It was a missed, now fixing is done.

File: app/views/recipes/index.html.erb (hunk 2 above)
Comment (amrendrakind) [reply app/views/recipes/index.html.erb]: @Deepakdanger, same here, It was a missed, and now fixing is done.

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
### highlights
- Professional readme file ✔️
- No linter errors ✔️
- Descriptive pull request message ✔️
approval


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**